applications/dashboard/models/class.messagemodel.php

```diff
@@ -162,13 +168,13 @@ public function getMessagesForLocation($Location, $Exceptions = array('[Base]'),
 
                 if (in_array($MController, $Exceptions)) {
                     $Visible = true;
-                } elseif ($MApplication == $Application && $MController == $Controller && $MMethod == $Method)
+                } elseif ($MApplication == $Application && $MController == $Controller && $MMethod == $Method) {
                     $Visible = true;
-
-                if ($Visible && !self::inCategory($CategoryID, val('CategoryID', $Message), val('IncludeSubcategories', $Message))) {
-                    $Visible = false;
                 }
 
+                $Visible = $Visible && self::inCategory($CategoryID, val('CategoryID', $Message), val('IncludeSubcategories', $Message));
+                $Visible = $Visible && $Session->checkPermission('Vanilla.Discussions.View', true, 'Category', $category['PermissionCategoryID']);
```

Comment:
Did you test this on messages not in a category? If `$category` is `null` you're trying to access an array element that doesn't exist here and testing permission for no reason.